tests/serial/three-solvers/ThreeSolversFirstParticipant.xml

```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<precice-configuration>
+  <solver-interface dimensions="2">
+    <data:scalar name="DataA" />
+    <data:scalar name="DataB" />
+
+    <mesh name="MeshOne">
+      <use-data name="DataA" />
+    </mesh>
+
+    <mesh name="MeshTwoA">
+      <use-data name="DataA" />
+    </mesh>
+
+    <mesh name="MeshTwoB">
+      <use-data name="DataB" />
+    </mesh>
+
+    <mesh name="MeshThree">
+      <use-data name="DataB" />
+    </mesh>
+
+    <participant name="SolverOne">
+      <use-mesh name="MeshOne" provide="yes" />
+      <write-data name="DataA" mesh="MeshOne" />
+    </participant>
+
+    <participant name="SolverTwo">
+      <use-mesh name="MeshOne" from="SolverOne" />
+      <use-mesh name="MeshTwoA" provide="yes" />
+      <use-mesh name="MeshTwoB" provide="yes" />
+      <read-data name="DataA" mesh="MeshTwoA" />
+      <write-data name="DataB" mesh="MeshTwoB" />
+      <mapping:nearest-neighbor
+        direction="read"
+        from="MeshOne"
+        to="MeshTwoA"
+        constraint="consistent" />
+    </participant>
+
+    <participant name="SolverThree">
+      <use-mesh name="MeshTwoB" from="SolverTwo" />
+      <use-mesh name="MeshThree" provide="yes" />
+      <read-data name="DataB" mesh="MeshThree" />
+      <mapping:nearest-neighbor
+        direction="read"
+        from="MeshTwoB"
+        to="MeshThree"
+        constraint="consistent" />
+    </participant>
+
+    <m2n:sockets from="SolverOne" to="SolverTwo" />
+    <m2n:sockets from="SolverTwo" to="SolverThree" />
+
+    <coupling-scheme:serial-explicit>
+      <participants first="SolverOne" second="SolverTwo" />
+      <max-time value="6" />
+      <time-window-size method="first-participant" />
+      <exchange data="DataA" mesh="MeshOne" from="SolverOne" to="SolverTwo" />
+    </coupling-scheme:serial-explicit>
+
+    <coupling-scheme:serial-explicit>
+      <participants first="SolverTwo" second="SolverThree" />
+      <max-time value="6" />
+      <time-window-size method="first-participant" />
+      <exchange data="DataB" mesh="MeshTwoB" from="SolverTwo" to="SolverThree" />
+    </coupling-scheme:serial-explicit>
```

Comment:
I just stumbled here: I think if this case is allowed for a compositional coupling scheme, then the assumption that `hasTimeWindowSize()` and `solverSetsTimeWindowSize()` are inverse is wrong. Both cases can coexist, because SolverTwo has a time window size (received from SolverOne) and at the same time it needs to set the time window size (for SolverThree).

This also has implications for the compositional coupling scheme: I'm not sure that we can simply iterate over all coupling schemes and apply a simple operation like `any_of` or `all_of`. Providing a public method (i.e. reachable from outside the coupling scheme) that is named `solverSetsTimeWindowSize()´ does not make sense for the compositional coupling scheme, because only individual `BaseCouplingSchemes` can set the time window size.